#With no hidden layer

In [0]:
import numpy as np

weight = np.array([0.7,0.2,-0.5,])
alpha = 0.1


In [0]:
streetLight = np.array([[0,0,1],
                        [0,1,1],
                        [0,0,1],
                        [1,1,1],
                        [0,1,1],
                        [1,0,1]])

# Stop = 0
# Walk = 1

walkStop = np.array([0,1,0,1,1,0])

In [31]:
print(len(streetLight))

6


In [32]:
for epoch in range(40):
  total_error = 0
  
  print("Epoch :", epoch+1)
  print("----------------------------------")
  print(walkStop)
  print("[", end = " ")

  for eachRow in range(len(walkStop)):
    input = streetLight[eachRow]
    target = walkStop[eachRow]

    # simple one layer prediction algo
    prediction = input.dot(weight)
    error = (target - prediction) ** 2
    total_error +=error
    delta = prediction - target 
    weight = weight - (alpha * input * delta)
    #print("Prediction for " + str(eachRow+1) + " : " + str(round(prediction,2)),end = ", "),
    print(round(prediction,2), end = ", ")

  print("]")

  print("Weight : " , weight)
  print("Error : ", error )

  print("----------------------------------")

Epoch : 1
----------------------------------
[0 1 0 1 1 0]
[ -0.5, -0.25, -0.32, 0.73, 0.09, 0.55, ]
Weight :  [ 0.67151  0.44315 -0.22959]
Error :  0.30514576
----------------------------------
Epoch : 2
----------------------------------
[0 1 0 1 1 0]
[ -0.23, 0.24, -0.13, 1.07, 0.39, 0.6, ]
Weight :  [ 0.60405911  0.57337347 -0.12345478]
Error :  0.36090706600849853
----------------------------------
Epoch : 3
----------------------------------
[0 1 0 1 1 0]
[ -0.12, 0.46, -0.06, 1.18, 0.54, 0.56, ]
Weight :  [ 0.52984133  0.65522424 -0.07978233]
Error :  0.3164892230726929
----------------------------------
Epoch : 4
----------------------------------
[0 1 0 1 1 0]
[ -0.08, 0.58, -0.03, 1.2, 0.63, 0.5, ]
Weight :  [ 0.45990131  0.71393979 -0.06005476]
Error :  0.249808223652252
----------------------------------
Epoch : 5
----------------------------------
[0 1 0 1 1 0]
[ -0.06, 0.66, -0.02, 1.19, 0.69, 0.43, ]
Weight :  [ 0.39744875  0.75977371 -0.04968233]
Error :  0.188971067261

## Error improved from 0.305 to 0.00000344
## Previous Weight [0.7,0.2,-0.5]
## Updated Weight [0.00262632  0.99981076 -0.00114148]
## Reflecting third value in every row is useless in this training process